.github/workflows/smoke-tests.yml

```diff
@@ -0,0 +1,105 @@
+name: "System Smoke Tests"
+on:
+  push:
+    # run on pushes to master or release/*
+    branches:    
+      - master
+      - release/*
+  pull_request:
+    # run on pull requests targeting master
+    branches:
+      - master
+jobs:
+  setup:
+    runs-on: self-hosted
+    steps:
+    - name: Check Test Cluster
+      run: |
+        if kubectl get nodes; then
+          kubectl get nodes
+        else
+          echo "test cluster not active. Attempting to create through kind..."
+          kind delete cluster 2>/dev/null
+          kind create cluster
+          kubectl get nodes
```

Comment:
need && between these, otherwise I'm fairly certain this step won't fail if a middle command fails.

Reply:
GitHub Actions will fail if any command fails (just re-tested to be sure)